students/toddschultz/session04/mailroom2.py

```diff
@@ -0,0 +1,133 @@
+    if person == "quit" or person == "q":
+        print ("Have a nice day!")
+    elif person == "list":
+        try_again()
+    donation = float(input("How much was the donation? "))
+    for i in donors:
+        if [i] == person:
+            i.append(donation)
+            d = dict()
+            d['name'] = i[0].title()
+            d['amount'] = donation
+            d['total'] = (sum(i[1:]))
+            d['donations'] = (len(i) - 1)
+            print("\n{name}\n".format(**d))
+            print("\tThank you very much for your generous donation. Your")
+            print("${amount:.2f} donation will allow us to continue our efforts.".format(**d))
+            print ("You have donated {donations} times totalling ${total:.2f}; AWESOME!!!".format(**d))
+            print("Our charity would not exist without your support.\n")
+            print("Sincerely:\n\nLeadership Team at Charity X.\n\n")
+            menu_selection(main_prompt, main_dispatch)
+    else:
+        new = (input("That looks like a new donor. Would you like to add a new donor? (Y)es or (N)o? ").lower())
+        if new == "yes" or new == "y":
+            donors.append([person])
+            donors[-1].append(donation)
+            print(donors)
+            d = dict()
+            d['name'] = person
+            d['amount'] = donation
+            print("\n{name}\n".format(**d))
+            print("\tThank you very much for your generouse donation. Your")
+            print("${amount:.2f} donation will allow us to continue our efforts.".format(**d))
+            print("Our charity would not exist without your support.\n")
+            print("Sincerely:\n\nLeadership Team at Charity X.\n\n")
+            menu_selection(main_prompt, main_dispatch)
+        else:
+            menu_selection(main_prompt, main_dispatch)
+def fun2(): #Report
+    donors_list = []
+    z = 0
+    print(donors)
+    for i in donors:
+        donors_list.append([i[0]]) #name
+        donors_list[z].append(sum(i[1:])) #total
+        donors_list[z].append(len(i)) #number + 1
+        aa=sum(i[1:])
+        donors_list[z].append(aa/len(i)) #average
+        z += 1
+    print("\nDonor Name\t\t| Total Given | Num Gifts | Average Gift")
+    print("----------------------------------------------------------------")
+    donors_list.sort(key = lambda x: int(x[1]),reverse=True)
+    for i in donors_list:
+        print("{:<22}".format(i[0]).title(), "  $", "{:11.2f}".format(i[1]), "\t\t{:<2}".format(i[2]-1), "{:2}".format("$"), "{:10.2f}".format(i[3]))
+    print("\n")
+    menu_selection(main_prompt, main_dispatch)
+
+def fun3(): #Letters
+    donors_list = []
+    z = 0
+    for i in donors:
+        donors_list.append([i[0]]) #name
+        donors_list[z].append(sum(i[1:])) #total
+        donors_list[z].append(len(i)) #number + 1
+        aa=sum(i[1:])
+        donors_list[z].append(aa/len(i)) #average
```

Comment:
i see same block of code above? could this be extracted into a function for code reuse?